database/sproc/payments.sql

```diff
@@ -4,78 +4,112 @@
+CREATE PROCEDURE `createPayment` (IN `sender_id` VARCHAR(255), IN `receiver_id` VARCHAR(255), IN `creator_id` VARCHAR(255), IN `parent_expense_id` INT(8), IN `currency_id` INT(8), IN `is_paid` BOOLEAN, IN `is_settled` BOOLEAN,  IN `amount` decimal(10, 2))
 BEGIN
 
-INSERT INTO `pending_payments` (`fk_sender_id`, `fk_receiver_id`, `fk_creator_id`, `fk_parent_expense_id`, `fk_currency_id`, `amount`)
-	VALUES (`sender_id`, `receiver_id`, `creator_id`, `parent_expense_id`, `currency_id`, `amount`);
+INSERT INTO `payment` (`fk_sender_id`, `fk_receiver_id`, `fk_creator_id`, `fk_parent_expense_id`, `fk_currency_id`, `is_paid`, `is_settled`, `amount`)
+	VALUES (`sender_id`, `receiver_id`, `creator_id`, `parent_expense_id`, `currency_id`, `is_paid`, `is_settled`, `amount`);
 
 END$$
 
 DELIMITER ;
 
 DELIMITER $$
 USE `doughBros_db`$$
-CREATE PROCEDURE `getAllPaymentsToUserInGroup` (IN `user_id` INT(8), IN `group_id` INT(8))
+CREATE PROCEDURE `getAllPaymentsByGroupExpense` (IN `parent_expense_id` INT(8))
 BEGIN
 
-SELECT * FROM `pending_payments` WHERE `user_id` = `fk_receiver_id`;
+SELECT * FROM `payment` WHERE `fk_parent_expense_id` = `parent_expense_id`;
 
 END$$
 
 DELIMITER ;
 
 DELIMITER $$
 USE `doughBros_db`$$
-CREATE PROCEDURE `getAllPaymentsFromUserInGroup` (IN `user_id` INT(8), IN `group_id` INT(8))
+CREATE PROCEDURE `getAllPendingPaymentsByGroupExpense` (IN `parent_expense_id` INT(8))
 BEGIN
 
-SELECT * FROM `pending_payments` WHERE `user_id` = `fk_sender_id`;
+SELECT * FROM `payment` WHERE (`fk_parent_expense_id` = `parent_expense_id` AND `is_paid` = 0 AND `is_settled` = 0);
 
 END$$
 
 DELIMITER ;
 
 DELIMITER $$
 USE `doughBros_db`$$
-CREATE PROCEDURE `payPayment` (IN `payment_id` INT(8))
+CREATE PROCEDURE `getAllPaidPaymentsByGroupExpense` (IN `parent_expense_id` INT(8))
 BEGIN
 
-BEGIN TRANSACTION;
-INSERT INTO `paid_payments`
-SELECT * FROM `pending_payments`
-WHERE `payment_id` = `payment_id`;
+SELECT * FROM `payment` WHERE (`fk_parent_expense_id` = `parent_expense_id` AND `is_paid` = 1 AND `is_settled` = 0);
+
+END$$
+
+DELIMITER ;
 
-DELETE FROM `pending_payments`
-WHERE `payment_id` = `payment_id`;
+DELIMITER $$
+USE `doughBros_db`$$
+CREATE PROCEDURE `getAllSettledPaymentsByGroupExpense` (IN `parent_expense_id` INT(8))
+BEGIN
 
-COMMIT;
+SELECT * FROM `payment` WHERE (`fk_parent_expense_id` = `parent_expense_id` AND `is_settled` = 1);
 
 END$$
 
 DELIMITER ;
 
 DELIMITER $$
 USE `doughBros_db`$$
-CREATE PROCEDURE `settlePayment` (IN `payment_id` INT(8))
+CREATE PROCEDURE `getAllPaymentsToUserInGroup` (IN `user_id` VARCHAR(255), IN `group_id` INT(8))
 BEGIN
 
-BEGIN TRANSACTION;
-INSERT INTO `settled_payments`
-SELECT * FROM `paid_payments`
-WHERE `payment_id` = `payment_id`;
+SELECT * FROM `payment` WHERE `fk_user_id` = `fk_receiver_id`;
+
+END$$
+
+DELIMITER ;
+
+DELIMITER $$
+USE `doughBros_db`$$
+CREATE PROCEDURE `getAllPaymentsFromUserInGroup` (IN `user_id` VARCHAR(255), IN `group_id` INT(8))
```

Comment:
@mebesius  brought up a good point about the getAllPaymentsToUserInGroup and getAllPaymentsFromUserInGroup sprocs in payments.sql, some copy pasta occurred here that was never completed, I'm gonna go fix that first before merge

- [ ] fix getAllPaymentsToUserInGroup
- [ ]  fix getAllPaymentsFromUserInGroup

Reply:
Think commit 6bf3a1d should fix it? Was a bit of a pain, but seemed to work when tested in workbench